### PR TITLE
`exp init`: UX updates

### PR DESCRIPTION
**Pull request by skshetry**
Now, DVC generates the boilerplate structure for all the dependencies that don't already exist. Also, the output structure for prompts was changed.

### CODE REVIEW

File: content/docs/command-reference/exp/init.md [hunk 1]
```diff
@@ -21,8 +21,10 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
 `dvc exp init` helps you get started with DVC Experiments quickly. It reduces
 boilerplate DVC procedures by creating a `dvc.yaml` file that assumes standard
 locations of your input data, <abbr>parameters</abbr>, source code, models,
-<abbr>metrics</abbr> and [plots](/doc/command-reference/plots). These locations
-can be customized through the [options](#options) below or via
+<abbr>metrics</abbr> and [plots](/doc/command-reference/plots). It also
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:24]: `<abbr>metrics</abbr>` doesn't seem to be working. Should `abbr` be working and supported for both `metrics` and `plots` here @jorgeorpinel @iesahin?

File: content/docs/command-reference/exp/init.md [hunk 2]
```diff
@@ -21,8 +21,10 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
 `dvc exp init` helps you get started with DVC Experiments quickly. It reduces
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:21]: Suggested rewording:

> It creates an experiment pipeline in a `dvc.yaml` file, which assumes standard locations for the inputs (source code, data, and parameters) and outputs (model, metrics, and plots) of your model training command. It also generates the input locations if they don't exist.

File: content/docs/command-reference/exp/init.md [hunk 3]
```diff
@@ -155,23 +150,15 @@ The easiest route is using interactive mode and answering a few questions:
...
 Path to a code file/directory [src, n to omit]: src/train.py
 Path to a data file/directory [data, n to omit]: data/features
-Path to a model file/directory [models, n to omit]: models/predict.h5
 Path to a parameters file [params.yaml, n to omit]:
+
+Enter experiment outputs.
+Path to a model file/directory [models, n to omit]: models/predict.h5
 Path to a metrics file [metrics.json, n to omit]:
 Path to a plots file/directory [plots, n to omit]: n
 ...
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:164]: Now that the output is much shorter, can we include it all instead of truncating it?

Comment (dberenbaum) [reply content/docs/command-reference/exp/init.md:164]: `- epochs` should also be removed in the `dvc.yaml` below.

File: content/docs/command-reference/exp/init.md (hunk 1 above)
Comment (daavoo) [reply content/docs/command-reference/exp/init.md:24]: It looks like I never started the initial P.R. #2644

Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:24]: Thanks for the reminder. It'll work eventually 😬

File: content/docs/command-reference/exp/init.md [hunk 4]
```diff
@@ -21,8 +21,10 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
 `dvc exp init` helps you get started with DVC Experiments quickly. It reduces
 boilerplate DVC procedures by creating a `dvc.yaml` file that assumes standard
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:22]:
```suggestion
repetitive DVC procedures by creating a `dvc.yaml` file that assumes standard
```

File: content/docs/command-reference/exp/init.md [hunk 5]
```diff
@@ -21,8 +21,10 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
...
 locations of your input data, <abbr>parameters</abbr>, source code, models,
-<abbr>metrics</abbr> and [plots](/doc/command-reference/plots). These locations
-can be customized through the [options](#options) below or via
+<abbr>metrics</abbr> and [plots](/doc/command-reference/plots). It also
+generates boilerplate structure for your input data, parameters and source code
+if they don't already exist. These locations can be customized through the
```
Comment (jorgeorpinel) [inline content/docs/command-reference/exp/init.md:26]:
```suggestion
<abbr>metrics</abbr> and [plots](/doc/command-reference/plots)
(generating the boilerplate structure if needed).
These locations can be customized through the
```

**Review by jorgeorpinel — COMMENTED** [review]
Committing 👇🏼

File: content/docs/command-reference/exp/init.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:21]: I like this @dberenbaum . I would make some edits though. Let's move it to a separate PR I think... ⌛

**Comment by jorgeorpinel** [comment]
p.s. does https://dvc.org/doc/user-guide/experiment-management/experiments-overview#initialize-dvc-experiments-on-any-project need an update?

**Comment by jorgeorpinel** [comment]
I'll take this one over.

File: content/docs/command-reference/exp/init.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md]: Added full output. Removed `params` group from dvc.yaml sample.

File: content/docs/command-reference/exp/init.md [hunk 6]
```diff
@@ -19,10 +19,11 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
 ## Description
 
 `dvc exp init` helps you get started with DVC Experiments quickly. It reduces
-boilerplate DVC procedures by creating a `dvc.yaml` file that assumes standard
+repetitive DVC procedures by creating a `dvc.yaml` file that assumes standard
 locations of your input data, <abbr>parameters</abbr>, source code, models,
-<abbr>metrics</abbr> and [plots](/doc/command-reference/plots). These locations
-can be customized through the [options](#options) below or via
+<abbr>metrics</abbr> and [plots](/doc/command-reference/plots) (generating the
+boilerplate structure if needed). These locations can be customized through the
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:25]: > (generating the
boilerplate structure if needed)

This is only true of the inputs, which is one reason I suggested rewording the intro above to specify "inputs" and "outputs" (and why I think it's hard to separate #3440).

File: content/docs/command-reference/exp/init.md [hunk 7]
```diff
@@ -191,8 +182,6 @@ train:
   deps:
     - data/features
     - src/train.py
-  params:
-    - epochs
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:195]: We should be keeping `params:` but dropping `epochs`. So it should look like:

```yaml
train:
  cmd: python src/train.py
  deps:
    - data/features
    - src/train.py
  params:
  outs:
    - models/predict.h5
  metrics:
    - metrics.json:
        cache: false
```

File: content/docs/command-reference/exp/init.md [hunk 8]
```diff
@@ -155,26 +149,23 @@ The easiest route is using interactive mode and answering a few questions:
...
-└── src
-
+Enter experiment dependencies.
 Path to a code file/directory [src, n to omit]: src/train.py
 Path to a data file/directory [data, n to omit]: data/features
-Path to a model file/directory [models, n to omit]: models/predict.h5
 Path to a parameters file [params.yaml, n to omit]:
+
+Enter experiment outputs.
+Path to a model file/directory [models, n to omit]: models/predict.h5
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:160]: This could be in a separate PR, but we should probably use `predict.h5` or `model.h5` instead of `models/predict.h5` until https://github.com/iterative/dvc/issues/5802 is resolved.

File: content/docs/command-reference/exp/init.md (hunk 6 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:25]: > This is only true of the inputs

What about metrics.json, models/ , and plots/ @dberenbaum ? Sure, it's up to the user where to output any other artifact. Should we just note that?

Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:25]: Oh I see. Those are only created with in `-i` mode... ⌛

Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md]: OK I changed the Description now. Actually there's even more nuance because with `-i` mode only the locations specified are created. Even input locations can be omitted and thus not generated. PTAL

File: content/docs/command-reference/exp/init.md (hunk 7 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md]: That's the default now? OK, done.

File: content/docs/command-reference/exp/init.md (hunk 8 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:172]: Not sure. a) it's a bug I think (shouldn't doc based on that) and b) the output of `exp init` does say 

```
Ensure your experiment command creates ... models/predict.h5.
```

Comment (dberenbaum) [reply content/docs/command-reference/exp/init.md:172]: Sorry, I should have clarified. This is what happens now:

```
$ dvc exp init -i
Command to execute: python src/train.py

Enter experiment dependencies.
Path to a code file/directory [src, n to omit]: src/train.py
'src/train.py' does not exist, the file will be created.
Path to a data file/directory [data, n to omit]: data/features
'data/features' does not exist, the directory will be created.
Path to a parameters file [params.yaml, n to omit]:
'params.yaml' does not exist, the file will be created.

Enter experiment outputs.
Path to a model file/directory [models, n to omit]: models/predict.h5
Path to a metrics file [metrics.json, n to omit]:
Path to a plots file/directory [plots, n to omit]: n

ERROR: unexpected error - [Errno 2] No such file or directory: '/private/tmp/tmprepo/models/.gitignore'

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
```

> it's a bug I think

Yeah, we need to prioritize fixing this in DVC.

Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md:172]: K. Marking this PR with `wait` status. Please lmk when to follow up if possible!

File: content/docs/command-reference/exp/init.md [hunk 9]
```diff
@@ -155,26 +161,23 @@ The easiest route is using interactive mode and answering a few questions:
...
 Path to a metrics file [metrics.json, n to omit]:
 Path to a plots file/directory [plots, n to omit]: n
-...
+
+Creating dependencies: src/train.py and params.yaml
+Creating train stage in dvc.yaml
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:177]:
```suggestion
Creating dependencies: src/train.py and params.yaml
Creating output directories: models
Creating train stage in dvc.yaml
```

File: content/docs/command-reference/exp/init.md [hunk 10]
```diff
@@ -18,38 +18,47 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
...
+A special deep learning `--type` of stage is supported (`dl`), which uses
 [DVCLive](/doc/dvclive) to monitor [checkpoints] during training of ML models.
 
-> `dvc exp init` is intended as a quick way to start running [DVC Experiments].
-> See the `dvc.yaml` specification for complex data pipelines.
+</admon>
+
+`dvc exp init` also generates parts of the boilerplate repo structure: only
+input files and directories by default, or all locations determined in
+interactive mode.
```
Comment (dberenbaum) [inline content/docs/command-reference/exp/init.md:54]:
```suggestion
`dvc exp init` also generates the boilerplate repo structure: input files and directories, 
and parent directories of outputs (the outputs themselves must be generated by the 
experiment command).
```

Comment (jorgeorpinel) [reply content/docs/command-reference/exp/init.md]: I rephrased this a little.

**Review by jorgeorpinel — APPROVED** [review]
LGTM but WDYT @dberenbaum ? Feel free to merge.

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
@jorgeorpinel there are some CI checks that failed

File: content/docs/command-reference/exp/init.md [hunk 11]
```diff
@@ -18,38 +18,46 @@ usage: dvc exp init [-h] [-q | -v] [--run] [--interactive] [-f]
...
+The only required argument is the terminal command that runs your experiment(s).
+It can be provided directly [as an argument](#the-command-argument) or by using
+the `--interactive` (`-i`) mode (which will prompt for it). The command will be
 wrapped as a <abbr>stage</abbr> that `dvc exp run` can execute.
 
-Different types of stages are supported, such as `dl` (deep learning) which uses
+<admon type="tip">
+
+A special deep learning `--type` of stage is supported (`dl`), which uses
 [DVCLive](/doc/dvclive) to monitor [checkpoints] during training of ML models.
```
[the remark this answers is not in the document]
Comment (dberenbaum) [reply content/docs/command-reference/exp/init.md:47]: Or we could get rid of this tip.

**Comment by dberenbaum** [comment]
> LGTM but WDYT @dberenbaum ? Feel free to merge.

I saw one issue with `--type=dl` (which is no longer an option), but besides that and CI issues, LGTM. Thanks!

**Comment by jorgeorpinel** [comment]
> there are some CI checks that failed

There was something wrong in `master`. Recently fixed in #3640 -- should be fine but merging it in just in case...

**Comment by jorgeorpinel** [comment]
> one issue with --type=dl (which is no longer an option)

Then we missed a cmd ref update somewhere because it's still mentioned in a few places of https://dvc.org/doc/command-reference/exp/init
